apps/server/src/modules/authentication/controllers/dto/oauth-login.response.ts

```diff
@@ -0,0 +1,14 @@
+import { ApiPropertyOptional } from '@nestjs/swagger';
+import { LoginResponse } from './login.response';
+
+export class OauthLoginResponse extends LoginResponse {
+	@ApiPropertyOptional({
+		description: 'The external id token which is from the external oauth system.',
+	})
+	externalIdToken?: string;
```

Comment:
is the id token still optional now. I think we can get it with every Oauth2 login.

Reply:
Yes, it is. the idp only gives us the id_token when scope openid is available

Reply:
@alweber-cap can you decide which way we want to go pls

Reply:
what arne says. oauth2 is not oidc. the token is only requested with the proper scope and thus not always present (currently it is, but to be futureproof etc.)

Reply:
but we exclusively use oidc and also validate the id token

Reply:
imho we should change the validation in the flow in seperate ticket. We are only allowed to validate the id token when scope is there and of course when we got an id token from the idp. no matter which standard